addons/edi/models/edi_issues.py

```diff
@@ -34,54 +31,58 @@ class ProjectTask(models.Model):
                                       index=True, ondelete='cascade')
 
 
-class EdiIssue(models.AbstractModel):
-    """EDI Issue-Tracked Object
+class MainIssue(models.AbstractModel):
+    """Main Issue-Tracked Object"""
 
-    EDI errors are raised in an issue tracker (and logged as messages
-    on the originating object).
-    """
+    # To consider: Would be good to move this abstract class in a module that only depends
+    # on project module in the future. So if we want to replicate this solution not to depend on
+    # EDI module by force.
+    _name = "main.issues"
+    _description = "Main Issue Object"
 
-    _name = 'edi.issues'
-    _description = "EDI Issue-Tracked Object"
-    _inherit = ['mail.thread']
+    # Empty list variable, will be overriden in the specific models where used.
+    ISSUES_FIELD_MAP = []
 
-    def _default_project_id(self):
-        return self.env.ref('edi.project_default')
-
-    project_id = fields.Many2one('project.project', string="Issue Tracker",
-                                 required=True, default=_default_project_id)
-    issue_ids = fields.One2many('project.task', string="Issues",
-                                domain=['|', ('stage_id.fold', '=', False),
-                                        ('stage_id', '=', False)])
-    issue_count = fields.Integer(string="Issue Count",
-                                 compute='_compute_issue_counts', store=True)
-    rel_issue_count = fields.Integer(string="Related Issue Count",
-                                     compute='_compute_issue_counts',
-                                     store=True)
+    issue_ids = fields.One2many(
+        "project.task",
+        string="Issues",
+        domain=["|", ("stage_id.fold", "=", False), ("stage_id", "=", False)],
+    )
+    issue_count = fields.Integer(string="Issue Count", compute="_compute_issue_counts", store=True)
+    rel_issue_count = fields.Integer(
+        string="Related Issue Count", compute="_compute_issue_counts", store=True
+    )
 
-    @api.multi
-    @api.depends('issue_ids', 'issue_ids.stage_id', 'issue_ids.edi_doc_id',
-                 'issue_ids.edi_gateway_id', 'issue_ids.edi_transfer_id')
+    @api.model
+    def display_issue_counts_depends(self):
+        """
+        Main fields which _compute_issue_counts depends on, it will be extended with super call
+        on specific models where used.
+        """
+        return ["issue_ids", "issue_ids.stage_id"]
+
+    @api.depends(lambda self: self.display_issue_counts_depends())
     def _compute_issue_counts(self):
         """Compute number of open issues (for UI display)"""
-        inverse = self._fields['issue_ids'].inverse_name
-        related = [edi_field
-                   for field, edi_field in EDI_FIELD_MAP
-                   if edi_field != inverse and field not in self._fields]
+        inverse = self._fields["issue_ids"].inverse_name
+        related = [
```

Comment:
If we are decoupling this model from EDI we should probably use a different name than `edi_field`; `model_field` perhaps?

Reply:
Yes missed that